OSSM-11158 [DOCS] Update New features, enhancements, bug fixes, and known issues for 3.2 RNs (#101669)
openshift/openshift-docs: modules/ossm-release-notes-3-2-new-features-enhancements.adoc

// Module included in the following assemblies:
//
// * service-mesh-docs-main/ossm-release-notes/ossm-release-notes.adoc

:_mod-docs-content-type: REFERENCE
[id="ossm-release-3-2-new-features-enhancements_{context}"]
= {SMProductName} version 3.2 new features and enhancements

[role="_abstract"]
This release makes {SMProductName} 3.2 generally available, adds new features, addresses Common Vulnerabilities and Exposures (CVEs), and is supported on {ocp-product-title} 4.18 and later.

> For a list of supported component versions and support features, see "Service Mesh 3.2 feature support tables".

Author: @jewertow As per doc guidelines, the links are not supposed to be in modules. You can find all the highlighted links in the additional resources section: https://101669--ocpdocs-pr.netlify.app/openshift-service-mesh/latest/ossm-release-notes/ossm-release-notes.html#additional-resources

> When upgrading from {SMProduct} 2.x, first you must migrate to version 3.0. Then, you can upgrade to version 3.2. For more information see, "Migrating from Service Mesh 2 to Service Mesh 3".

Reviewer: They'll need to go through 3.0 -> 3.1 -> 3.2, so perhaps update the first sentence to: 
> When upgrading from {SMProduct} 2.x, first you must migrate to version 3.0 and then 3.1. Then, you can upgrade to version 3.2.

[id="general-availability-of-istio-ambient-mode_{context}"]
== General availability of Istio ambient mode

This enhancement brings the core features of {istio} ambient mode, ztunnel and waypoint, to general availability. Ambient mode reduces the resource costs of running a service mesh by removing the need for sidecar proxies and introducing a simplified architecture.

[id="Updated-feature-support-matrix-for-ambient-mode_{context}"]
== Updated feature support matrix for ambient mode

This enhancement provides an updated feature support matrix for {istio} ambient mode. Not all sidecar mode features are supported in ambient mode. For detailed information about supported and unsupported features, see "Service Mesh feature support tables".

[id="network-policy-updates-for-ambient-mode-traffic_{context}"]

> == Network policy updates for ambient mode traffic

Reviewer: ```suggestion
== Network policy updates may be required for ambient mode traffic
```

This enhancement introduces Layer 4 (L4) tunneling for ambient workloads. The `ztunnel` component tunnels pod traffic over Transmission Control Protocol (TCP) port `15008`. If existing {k8s} `NetworkPolicy` configurations block inbound traffic on this port, update them to allow inbound TCP traffic on port `15008` for ambient workloads. Sidecar workloads must also allow inbound traffic on this port to communicate with ambient workloads.
For more information, see "Configuring network policies for ambient mode".

[id="improved-probe-reliability-with-OVN-Kubernetes-local-gateway-mode_{context}"]

> == Improved probe reliability with OVN-Kubernetes local gateway mode

Reviewer 2: I'm afraid, this is not an improvement, its a requirement for kubelet health-check probes to work. 
@longmuir can you please suggest how to frame this requirement?

> This enhancement ensures liveness and readiness probes continue to function correctly for workloads running in {istio} ambient mode. To maintain correct probe behavior, you can enable OVN-Kubernetes local gateway mode by setting `routingViaHost: true` in the `gatewayConfig` specification.

Reviewer: ```suggestion
To ensure liveness and readiness probes continue to function correctly for workloads running in {istio} ambient mode, you must enable OVN-Kubernetes local gateway mode by setting `routingViaHost: true` in the `gatewayConfig` specification.
``` 
Removing the wording that this is an enhancement and just stating it as a fact for Istio ambient mode that users should be aware of.

Reviewer 2: This looks good, thanks Jamie.

[id="general-availability-of-istio-ambient-mode_{context}"]
== General availability of Istio ambient mode

This enhancement brings the core features of {istio} ambient mode, ztunnel and waypoint, to general availability. Ambient mode reduces the resource costs of running a service mesh by removing the need for sidecar proxies with a new data plane architecture that consists of the following two levels of proxy:

* The node level ztunnel proxy

* The application focused waypoint proxy

[id="Updated-feature-support-matrix-for-ambient-mode_{context}"]
== Updated feature support matrix for ambient mode

This enhancement provides an updated feature support matrix for {istio} ambient mode. Not all sidecar mode features are supported in ambient mode. For detailed information about supported and unsupported features, see "Service Mesh feature support tables".

[id="network-policy-updates-for-ambient-mode-traffic_{context}"]
== Network policy updates may be required for ambient mode traffic

> {istio} ambient mode uses Layer 4 (L4) tunneling for traffic between workloads. The `ztunnel` component tunnels pod traffic over TCP port `15008`. If existing {k8s} `NetworkPolicy` configurations block inbound traffic on this port, update them to allow inbound TCP traffic on port `15008` for ambient workloads. Sidecar workloads must also allow inbound traffic on this port to communicate with ambient workloads.

Reviewer 3: "ambient mode uses Layer 4 (L4) tunneling for traffic between workloads" - this is not quite correct. The payload being tunneled is TCP (L4), but the tunnel itself is implemented at the application layer (L7) using HTTP CONNECT, so I would say:
```suggestion
{istio} ambient mode uses an application-layer tunnel (L7) - called HBONE - to carry TCP traffic (L4) securely between workloads. The `ztunnel` component tunnels pod traffic over TCP port `15008`. If existing {k8s} `NetworkPolicy` configurations block inbound traffic on this port, update them to allow inbound TCP traffic on port `15008` for ambient workloads. Sidecar workloads must also allow inbound traffic on this port to communicate with ambient workloads.
```
